File: AppRemover/src/main/java/com/vs_unusedappremover/common/HandlerTimer.java
package com.vs_unusedappremover.common;

import android.os.Handler;

public abstract class HandlerTimer {
	
	private final Handler handler;
	private final long intervalMillis;
	private boolean isEnabled;
	
	public HandlerTimer(Handler handler, long intervalMillis) {
		this.handler = handler;
		this.intervalMillis = intervalMillis;
	}
	
	public void setEnabled(boolean value) {
		isEnabled = value;
		if (isEnabled) {
			timerRunnable.run();
		}
	}
	
	protected abstract void onTick();
	
	private final Runnable timerRunnable = new Runnable() {		
		@Override
		public void run() {
			onTick();
			if (isEnabled) {
				handler.removeCallbacks(this);
				handler.postDelayed(this, intervalMillis);
			}
		}
	};
}

File: AppRemover/src/main/java/com/vs_unusedappremover/common/HandlerTimer.kt
package com.vs_unusedappremover.common

import android.os.Handler

abstract class HandlerTimer(private val handler: Handler, private val intervalMillis: Long) {
    private var isEnabled: Boolean = false

    fun setEnabled(value: Boolean) {
        isEnabled = value
        if (isEnabled) {
            timerRunnable.run()
        }
    }

    protected abstract fun onTick()

    private val timerRunnable = object : Runnable {
        override fun run() {
            onTick()
            if (isEnabled) {
                handler.removeCallbacks(this)
                handler.postDelayed(this, intervalMillis)
            }
        }
    }
}

